src/java/org/apache/cassandra/cql3/restrictions/RestrictionSet.java

```diff
@@ -104,6 +104,17 @@ public List<ColumnMetadata> getColumnDefs()
     {
         return new ArrayList<>(restrictions.keySet());
     }
+    
+    @Override
+    public Collection<ColumnMetadata> getColumnDefinitions()
```

Comment:
Why not use the already existing `getColumnDefs()`?

Reply:
I think the worry there (for performance) was what appeared to be a needless defensive copy in `getColumnDefs()`. I didn't want to audit the 12 other existing callers of `getColumnDefs()`. Certainly open to naming changes...

Reply:
Also, we may be able to delete these entirely, depending on what we do w/ https://github.com/apache/cassandra/pull/2105#discussion_r1083887949

Reply:
Ohhh icwym with the defensive copy. Then I would either document it, rename it or add a flag to choose the desired behavior. Anything that makes it obvious why there are 2 options to choose.

Reply:
Agreed. Will at least JavaDoc that for posterity...